woocommerce/woocommerce-ios
# [Issue Refunds] Add GatewayPayment storage structure

part of #3110 
fix https://github.com/woocommerce/woocommerce-ios/issues/3019

# Why

This PR does two things(sorry)

1. As a continuation of adding `PaymentGateway` networking support this PR takes care of adding the necessary `Storage` classes so we can use it(`PaymentGateway`) in our [future Yosemite actions and store](https://github.com/woocommerce/woocommerce-ios/pull/3141)

2. While adding a `Transformable` property on `Storage.PaymentGateway` I noticed that it generated a new warning because now, by default, those types of properties use an insecure NSCoding value transformer. This PR fixes that warning for all other transformable properties in the project.

# How

1. 
- Added Model 36 that includes `PaymentGateway` Core Data Entity
- Added auto-gen `+CoreDataProperties` and `+CoreDataClass` extensions
- Updated `StorageType+Extensions` to provide functions to load payment gateways from storage.
- Added `ReadOnlyConvertible` methods to `PaymentGateway`

2.
- Update `transformable` properties on Product, to use `NSSecureUnarchiveFromData` as the property value transformer instead of the default(unsecure) one.

# Testing Steps

1. Not much to test here, but you can see it working on [this PR](https://github.com/woocommerce/woocommerce-ios/pull/3141)

2. Make sure that after compiling(cleaning it first) the project, no more core data warnings are present

Update release notes:
